sqlglot/dialects/tsql.py

```diff
@@ -502,6 +508,110 @@ class Parser(parser.Parser):
         STRING_ALIASES = True
         NO_PAREN_IF_COMMANDS = False
 
+        OPTION_PREFIX_TO_PARSE_FUNCTION = {
+            "LABEL": lambda self: self._generic_parse_option(
+                "LABEL",
+                [
+                    lambda: self._match(TokenType.EQ)
+                    and self._return_option_with_value("LABEL", self._parse_primary())
+                ],
+            ),
```

Comment:
Can we refactor this map definition to simplify it a bit? One idea is to fetch (e.g., `LABEL`) the key using `self._prev.text.upper()` instead of repeating it in the first arg. I implemented something similar in another [PR](https://github.com/tobymao/sqlglot/pull/3027/files#diff-14ef9d8b99f48dcf8f6a4e3bbab7bf55a9d85d2b83d684c3c822f0f94fcd1d07R493).